chore: Hyperv and upstream Containerd package support (#3688)
Azure/aks-engine: docs/topics/features.md

```json
      "kubernetesConfig": {
        "networkPlugin": "azure",
        "containerRuntime": "containerd",
        "windowsContainerdURL": "..."
      }
```

### Hyper-v support
This feature in AKS-Engine is for testing the in-development versions of ContainerD and Kubernetes, and is not for production use. Be sure to review [open issues](https://github.com/azure/aks-engine/issues?q=containerd+label%3Awindows+is%3Aopen) if you want to test or contribute to this effort.

> The current default for hyper-v sets hyper-v containers as default and adds a process isolated option for the currently selected host VM OS version. If you VM OS version is Windows Server 2004 (10.0.19041) and you apply a pod spec with now updates you will get a 2004 container running in hyper-v container.  

Reviewer: ```suggestion
The current default for Hyper-V sets Hyper-V containers as default and adds a process isolated option for the currently selected host VM OS version. If your VM OS version is Windows Server 2004 (10.0.19041) and you apply a pod spec with new updates, you will get a 2004 container running in a Hyper-V container.  
```

> If you wish to have process isolated or OS version below your current Host OS version, you will need to create a RuntimeClass object and map the pod to the RuntimeClass.  Note that Hyper-v support is currently backwards compatible.  You have to have a Host OS that is the same version or newer than the version of the container you wish to run.  Multi-arch container images are not supported. You must have a single arch image if hyperv is enabled in containerd.

Reviewer: ```suggestion
If you wish to have process isolated or use an OS version below your current Host OS version, you will need to create a RuntimeClass object and map the pod to the RuntimeClass.  Note that Hyper-V support is currently backwards compatible.  You have to have a Host OS that is the same version or newer than the version of the container you wish to run.  Multi-arch container images are not supported. You must have a single arch image if Hyper-V is enabled in containerd.
```

metadata:
  name: windows-2019
handler: 'runhcs-wcow-hypervisor-17763'
scheduling:
  nodeSelector:
    kubernetes.io/os: 'windows'
    kubernetes.io/arch: 'amd64'
    node.kubernetes.io/windows-build: '10.0.19041'
  tolerations:
  - effect: NoSchedule
    key: os
    operator: Equal
    value: "windows"
```

And then you would be able to run a 2019/1809 (10.0.17763) container by setting the `runtimeClassName` to the `windows-2019` RuntimeClass on the container template:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: iis-ltsc2019
  labels:
    app: iis-ltsc2019
spec:
  replicas: 1
  template:
    metadata:
      name: iis-ltsc2019
      labels:
        app: iis-ltsc2019
    spec:
      runtimeClassName: windows-2019
      containers:
      - name: iis
        image: mcr.microsoft.com/windows/servercore/iis:windowsservercore-ltsc2019
        resources:
          limits:
            cpu: 1
            memory: 800m
          requests:
            cpu: .1
            memory: 300m
        ports:
          - containerPort: 80
      nodeSelector:
        "kubernetes.io/os": windows
  selector:
    matchLabels:
      app: iis-ltsc2019

```

Current limitations:

> - Currently the Runtime Hanlders are not configurable.  

Reviewer: ```suggestion
- Currently the Runtime handlers are not configurable.  
```

> - If you specific a hanlder that does not map the the fields in [../../parts/k8s/containerdhypervtemplate.toml](parts/k8s/containerdhypervtemplate.toml) then then the container will not start. 

Reviewer: ```suggestion
- If you specify a handler that does not map the fields in [../../parts/k8s/containerdhypervtemplate.toml](parts/k8s/containerdhypervtemplate.toml), then the container will not start. 
```

> - If you map to container version that is higher than your current OS image your container will not start.

Reviewer: ```suggestion
- If you map to a container version that is higher than your current OS image your container will not start.
```

- Multi-arch container images are not supported

You can learn more about RuntimeClasses and the future of the Windows support:

- https://kubernetes.io/docs/concepts/containers/runtime-class/
- https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/windows-runtimeclass-support.md

> ### Building ContainerD with Hyper-v

Reviewer: ```suggestion
### Building ContainerD with Hyper-V
```

> As of Aug 10, 2020, the ContainerD hyper-v support don't have public builds available. This repo has a script that will build it from source and create a ZIP file: [build-windows-containerd.sh](../../scripts/build-windows-containerd.sh)

Reviewer: ```suggestion
As of Aug 10, 2020, the ContainerD Hyper-V support doesn't have public builds available. This repo has a script that will build it from source and create a ZIP file: [build-windows-containerd.sh](../../scripts/build-windows-containerd.sh)
```
